trie/utils/verkle.go

```diff
@@ -47,28 +49,26 @@ var (
 )
 
 type PointCache struct {
-	cache map[string]*verkle.Point
-	lock  sync.RWMutex
+	cache *lru.Cache[string, *verkle.Point]
 }
 
 func NewPointCache() *PointCache {
+	// Each verkle.Point is 96 bytes.
+	verklePointSize := 96
+	capacity := maxPointCacheByteSize / verklePointSize
 	return &PointCache{
-		cache: make(map[string]*verkle.Point),
+		cache: lru.NewCache[string, *verkle.Point](capacity),
```

Comment:
`lru.Cache` differs from `lru.BasicLRU` only in the sense that `lru.Cache` is thread safe, which is a good idea if the prefetcher is around.